### PR TITLE
feat: add HTPipeSanitizeParams

### PR DESCRIPTION
**Pull request by VolodymyrWork23**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -653,6 +656,146 @@ export function HTPipeDoWork<
...
+>(left: TLeft, right: TRight) {
+  if (isHasSanitizeParams(left) && isHasSanitizeParams(right)) {
+    return {
+      sanitizeParams: (
+        context: TContextOutLeft extends TContextInRight
+          ? TContextInLeft
+          : TContextInRight & TContextInLeft
+      ) => {
+        const leftOut = left.sanitizeParams(context) || {};
+        const rightOut = right.sanitizeParams(leftOut) || {};
```
Comment (captaincaius) [inline src/index.ts:784]: minor... just return rightOut
- [ ] unneccessary expensive spread

File: src/index.ts [hunk 2]
```diff
@@ -653,6 +656,146 @@ export function HTPipeDoWork<
...
+  TContextOutRight extends ReturnType<TRight['sanitizeParams']>
+>(
+  left: TLeft,
+  right: TRight
+): HasSanitizeParams<
+  TContextInLeft &
+    Omit<
+      TContextInRight,
+      TContextOutLeft extends void ? keyof {} : keyof TContextOutLeft
+    >,
```
Comment (captaincaius) [inline src/index.ts:687]: the return side should be only TContextOutRight
- [ ] fix typescript-side snowballing on all sanitizers

Comment (captaincaius) [reply src/index.ts:687]: actually... i think the input-side is kind of not right too.  I think it should probably be only TContextInLeft ... It should be illegal to have a "right" that needs more than left provides.
- [ ] piper's return value's input side
- [ ] fix the type constraints above accordingly too (it should probably be simpler)

Comment (VolodymyrWork23) [reply src/index.ts:687]: done here: https://github.com/captaincaius/hipthrusts/pull/44/files

File: src/index.ts (hunk 1 above)
Comment (VolodymyrWork23) [reply src/index.ts:784]: done here: https://github.com/captaincaius/hipthrusts/pull/44/files

**Comment by captaincaius** [comment]
obsolete

**Pull request closed without merging**